### ISSUE TITLE
After switching tabs, useSupabaseClient methods stop working and don't return

### ISSUE THREAD

**Issue by ErwinAI** [nuxt-modules/supabase#273]
It seems like things stop working if you tab away or come back to the page moments later. Ive tried this both with retrieving from the Supabase db and auth and for both no promise is returned and it essentially gets stuck. Reproduction below is for the signOut (easiest of the two to reproduce).

This makes this module and supabase as a whole unusable for my app and prevents me from going to production.

### Version
@nuxtjs/supabase: v1.1.0 (latest)
nuxt: v3.7.3 (latest)

### Reproduction

```
const supabase = useSupabaseClient()
const signOut = async () => {
  const user = useSupabaseUser()

  console.log(user) // does log

  console.log(supabase) // does log

  const { error } = await supabase.auth.signOut()
  
  // no code is executed after this

  if(error) {

    console.log(error)

  } else {
    console.log('redirecting..')

    navigateTo('/account/login')
  }
}
```

This results in two console logs (see code comments):

<img width="482" alt="image" src="https://github.com/nuxt-modules/supabase/assets/2168688/aa1568e9-0e62-4ee7-9ea4-9a3ede866c45">

### Steps to reproduce

Login and switch tabs (perhaps wait for a bit or tab in/out of browser), then activate `signOut` function in the code above to sign out and notice that no redirection is happening and no log is shown in the console after the signOut function. It seems as if the promise is never returned?

### What is Expected?
That either an error is returned, or a fulfilled promise with `error: null` as per the documentation so that navigateTo is activated (see code above)

### What is actually happening?
The user is NOT logged out (reloading the page and checking the user still returns an authenticated user object) and no promise is returned.

**Comment by ErwinAI** [comment]
Additional information: No network requests are visible either.

**Comment by ErwinAI** [comment]
This issue (https://github.com/nuxt-modules/supabase/issues/220) is exactly the issue I'm running into. But it mentions that it was fixed in a previous version. How come I'm experiencing this again now 🤔

**Comment by ErwinAI** [comment]
Adding this to my package.json seems to work. Was supposed to be fixed in 1.0.1 though 🤔?

```
"overrides": {
  "@supabase/supabase-js": "2.31.0",
  "@supabase/gotrue-js": "2.43.1"
},
```

**Comment by jawngee** [comment]
Yes I just ran into this too.  For me, it was any MFA related calls like `client.auth.mfa.getAuthenticatorAssuranceLevel`

**Comment by lassidevs** [comment]
Hey Erwin, 

I'm experiencing the same issue, did you find a fix for this?

**Comment by timrekelj** [comment]
This same error is happening to me too. I tried it with the same versions of nuxt and nuxt/supabase as @ErwinAI and newer ones. I added overrides and it still did not help.

**Comment by lammerfalcon** [comment]
i have the same issue, after change tabs, or lose window focus and then go back and try do any request - nothing happens

**Comment by lammerfalcon** [comment]
> Adding this to my package.json seems to work. Was supposed to be fixed in 1.0.1 though 🤔?
> 
> ```
> "overrides": {
>   "@supabase/supabase-js": "2.31.0",
>   "@supabase/gotrue-js": "2.43.1"
> },
> ```

this solution doesent work for me(

**Comment by ErwinAI** [comment]
This is what I ended up doing and has worked for me.

`"overrides": {
    "@supabase/supabase-js": "2.31.0",
    "@supabase/gotrue-js": "2.43.1"
  },`

**Comment by nosizejosh** [comment]
Same issue with these versions, using yarn on windows.
```
"@nuxtjs/supabase": "^1.1.4",
"nuxt": "^3.8.2",
```
Only way to get it to work was to add 
```
"resolutions": {
    "@supabase/gotrue-js": "2.43.1"
  }

```
to my package.json file.

is there an update that really fixes this issue so we can be in sync?

**Comment by nosizejosh** [comment]
This issue seem to be resolved after updating to 1.1.6  and removing the resolution.
Can someone confirm this?

**Comment by arthurmeb** [comment]
> This issue seem to be resolved after updating to 1.1.6 and removing the resolution. Can someone confirm this?

Is there a 1.1.6 version out? Changelog says the latest release was 1.1.5 on December 8th.
I'm currently facing this issue

**Comment by edarioq** [comment]
Ohh so the fix is to use a `setTimeout` ? Meaning =>

```
 const {
      data: { subscription },
    } = supabase.auth.onAuthStateChange(async (_, session) => {
      if (session?.user) {
        setTimeout(async () => {
          // Call the database here !
        })
      } else {
        setUser(null);
      }
    });
```

**Comment by Renatello** [comment]
@ScreamZ, @edarioq thank you for the solution. I spent hours trying to figure out why this happens

**Comment by 136sergio** [comment]
> Ohh so the fix is to use a `setTimeout`:
> 
> ```
>  const { data: { subscription } } = supabase.auth.onAuthStateChange(async (_, session) => {
>   if (session?.user) {
>     setTimeout(async () => {
>       // Call database here !
>       const { data } = await supabase
>           .from('users')
>           .select('*')
>           .eq('id', session.user.id)
>           .single();
>     })
>   }
> });
> ```

Thank you so much, I had been testing for hours and your solution is the only thing that has worked.

**Comment by holys** [comment]
https://supabase.com/docs/reference/javascript/auth-onauthstatechange

**Comment by waqarilyas** [comment]
> setTimeout(async () => {
>       // Call database here !
>       const { data } = await supabase
>           .from('users')
>           .select('*')
>           .eq('id', session.user.id)
>           .single();
>     })

This worked, but can anyone explain why? @edarioq

**Comment by ByteSh0ck** [comment]
God i struggled so long with this. The setTimeout fixed it for me as well.

**Comment by edyyy19** [comment]
Thank you some much, ive been trying to solve this thing for such a long time. Thank you!

**Comment by Sjef200** [comment]
G

**Comment by M-Arabyat-Gasable** [comment]
> Adding this to my package.json seems to work. Was supposed to be fixed in 1.0.1 though 🤔?
> 
> ```
> "overrides": {
>   "@supabase/supabase-js": "2.31.0",
>   "@supabase/gotrue-js": "2.43.1"
> },
> ```

I can confirm that installing those versions worked for me, after changing the versions in package.json I used npm install, that solved it thanks

**Comment by NatsuD99** [comment]
> Ohh so the fix is to use a `setTimeout`:
> 
> ```
>  const { data: { subscription } } = supabase.auth.onAuthStateChange(async (_, session) => {
>   if (session?.user) {
>     setTimeout(async () => {
>       // Call database here !
>       const { data } = await supabase
>           .from('users')
>           .select('*')
>           .eq('id', session.user.id)
>           .single();
>     })
>   }
> });
> ```

My friend, you saved me from my 4 hours of hell. Also for people looking for this solution, make sure you are not using `async` function as callback

**Comment by ydt710** [comment]
@ScreamZ dude you are a fucking legend , i sat for 14 hours debuging this and found this , send me your btc wallet. I can't thank you enough

**Comment by ScreamZ** [comment]
> found

Pleased to read this in the morning ! Thanks mate 🙏🏻 this issue was insane !

`bc1qspx9unxe9exesh9hynuxx5ms8c8xfu3e86j83s`

**Comment by aindrei** [comment]
I still don't fully understand the problem, but the setTimeout solution worked.  Clearly there is a problem with accessing the  database inside of onAuthStateChange handler.
